### ISSUE TITLE
Make the "Discover services" URL configurable

### ISSUE THREAD

**Issue by Abbe98** [OpenRefine/OpenRefine#5604]
As someone deploying a multi-user OpenRefine instance I would like to be able to set a new URL for the "Discover services" button so that users can go to documentation regarding internal reconciliation endpoints rather than the Reconciliation service test bench to which anyone can add endpoints. 

### Proposed solution

Add a setting to OpenRefine allowing one to set a custom URL for the "Discover services" button.

### Alternatives considered

One option could be a setting to remove the button as a custom OpenRefine deployment can "pre-set" reconciliation services anyway. However, I could still see cases in which one would not like to "pre-set" all of ones services(multilingual services, etc).

**Comment by thadguidry** [comment]
Hmm, storing the setting in a preference doesn't seem ideal for deployment needs.  Needs to be a config file.
So maybe a preference variable read from the refine.ini or it's JAVA_OPTIONS?
Just wondering on the deployment side of things...seems like perhaps the refine.ini (a config file) is the way to make it easier for multi-user deployment?

**Comment by Abbe98** [comment]
@thadguidry yeah, refine.ini might be the better place. I do always end up making some configuration using settings anyway, but in this case it would probably be preferred if users can't edit it.

**Comment by thadguidry** [comment]
Ah, on that note... what configuration settings do you set and where?  Do we need to open those up to be read from refine.ini to help you more?

**Comment by wetneb** [comment]
I am not too sure about making this link configurable. As a user I would not suspect that the behaviour of this button could be different if I am using OpenRefine from some institutional deployment.

How about making the list of services initially registered in OpenRefine configurable instead? At the moment, my Wikidata recon endpoint is hardcoded, but that is quite bad (for instance because it could go down any time).

The advantage with this solution is that the user would not even have to register the services themselves by copying a URL into a form.

**Comment by Abbe98** [comment]
> Ah, on that note... what configuration settings do you set and where? Do we need to open those up to be optionally read from refine.ini to help you more?

@thadguidry I currently just set reconciliation endpoints by making a call to to the API which works very well.

> I am not too sure about making this link configurable. As a user I would not suspect that the behaviour of this button could be different if I am using OpenRefine from some institutional deployment.

@wetneb could it be that the button should look differently altogether to make the behavior more open-ended? It's quite confusing as it is given that it does not behave like the other buttons in the footer, and that it isn't styled like other external links in the reconciliation dialog.

> How about making the list of services initially registered in OpenRefine configurable instead? At the moment, my Wikidata recon endpoint is hardcoded, but that is quite bad (for instance because it could go down any time).
>
>The advantage with this solution is that the user would not even have to register the services themselves by copying a URL into a form.

This is in my opinion already solved as there is a setting which can be set with the API.

**Comment by Abbe98** [comment]
I currently remove the button altogether with some CSS but will likely move to an patch to change the URL instead as a multilingual reconciliation service I have follows the same pattern as the Wikidata one resulting in a lot of separate services. The original reason for removing the button is all because reconciliation services are a little bit of a security nightmare.

**Comment by thadguidry** [comment]
> How about making the list of services initially registered in OpenRefine configurable instead? At the moment, my Wikidata recon endpoint is hardcoded, but that is quite bad (for instance because it could go down any time).

@wetneb That was the idea I was driving at.  To set the endpoints made available in the UI by allowing them to be listed/setup in `refine.ini` to make it easier for administrators at institutions to deploy.  Basically, all such "settings" or "preferences" should be able to be read from a file (and maybe we just eventually move to a TOML file like other tools do for "config file" handling).
